plasma_framework/contracts/src/vaults/Vault.sol

```diff
@@ -21,10 +21,10 @@ contract Vault is Operated {
         zeroHashes = ZeroHashesProvider.getZeroHashes();
     }
 
-    modifier onlyFromExitGame() {
+    modifier onlyFromNonQuarantinedExitGame() {
         require(
-            framework.exitGameToTxType(msg.sender) != 0,
-            "Not called from a registered Exit Game contract"
+            ExitGameRegistry(framework).isSafeToUse(msg.sender),
```

Comment:
This is interesting. I believe it is not necessary to cast it to `ExitGameRegistry`. But I will leave this to you as your intention might be making this code explicit on the dependency and make code tracing easier.

Reply:
Yeah I saw possible name collision with Vaults as you suggested... but I think I'll leave it to satisfy the principle `explicit is better than implicit`

Reply:
ah...cool, didn't realized we can use this to avoid name crash. Now I like this casting practice actually.